packages/xrpl/src/models/transactions/common.ts

```diff
@@ -266,7 +394,7 @@ export interface GlobalFlags {}
 /**
  * Every transaction has the same set of common fields.
  */
-export interface BaseTransaction {
+export interface BaseTransaction extends Record<string, unknown> {
```

Comment:
Why do we need this carte blanche `extends Record<string, unknown>`? If the `BaseTransaction` (or) any of the child transaction need a key-value pair, that must be added explicitly.

This appears to go against the intent of this [PR (which focuses on refactoring `BaseRequest`)](https://github.com/XRPLF/xrpl.js/pull/2923)

Reply:
Let's move this discussion to #2966 if you have more thoughts to share